docs/manual/scala/guide/cluster/DomainModelling.md

```diff
@@ -1,19 +1,202 @@
+
+In Akka Typed, it's not possible to return an exception to the caller. All communication between the actor and the caller must be done via the `replyTo:ActorRef[R]` passed in the command. Therefore, if you want to signal a rejection, you most have it encoded in your reply protocol.
+
+```scala
+// Replies
+sealed trait Confirmation
+sealed trait Accepted extends Confirmation
+case object Accepted extends Accepted
+final case class Rejected(reason: String) extends Confirmation
+final case class CartState(items: Map[String, Int], status: String)
+
+// Commands
+sealed trait ShoppingCartCommand
+final case class UpdateItem(productId: String,
+                            quantity: Int,
+                            replyTo: ActorRef[Confirmation])
+  extends ShoppingCartCommand
+
+final case class Checkout(replyTo: ActorRef[Confirmation])
+  extends ShoppingCartCommand
+
+final case class Get(replyTo: ActorRef[CartState])
+```
+
+Note that we have different kinds of replies: `Confirmation` used when we want to modify the state. A modification request can be `Accepted` or `Rejected`. And `CartState` used when we want to read the state of the shopping cart. Keep in mind that `CartState` is not the shopping cart itself, but the representation we want to expose to the external world.
+
+### Modelling Events
+
+Next, we define the events that our model will persist. The events must extend Lagom's `AggregateEvent`. This is important for tagging events. We will cover this topic a little further.
+
+```scala
+sealed trait ShoppingCartEvent extends AggregateEvent[ShoppingCartEvent] {
+  def aggregateTag = ShoppingCartEvent.Tag
+}
+
+final case class ItemUpdated(productId: String, quantity: Int)
+  extends ShoppingCartEvent
+
+final case object CheckedOut extends ShoppingCartEvent
+```
+
+### Defining Commands Handlers
+
+Once you define your protocol in terms of Commands, Replies, Events and State, you need to specify the business rules of your model. The command handlers define how to handle each incoming command, which validations must be applied, and finally, which events will be persisted if any.
+
+You can encoding it in different ways. The [recommended style](https://doc.akka.io/docs/akka/2.6/typed/persistence-style.html#command-handlers-in-the-state) is to add the command handlers in your state classes. Since `ShoppingCart` have two state classes extensions, it makes sense to add the repective business rules validation on each state class. Each possible state will define how each command should be handled.
```

Comment:
```suggestion
You can encode it in different ways. The [recommended style](https://doc.akka.io/docs/akka/2.6/typed/persistence-style.html#command-handlers-in-the-state) is to add the command handlers in your state classes. Since `ShoppingCart` has two state class extensions, it makes sense to add the respective business rule validations on each state class. Each possible state will define how each command should be handled.
```

I got a bit confused with plurals. I've made a suggestion but I'm not sure it's correct either.